target/linux/qualcommax/image/ipq807x.mk

```diff
@@ -127,8 +127,9 @@ define Device/linksys_mx4200v1
 	IMAGE_SIZE := 147456k
 	NAND_SIZE := 512m
 	KERNEL_IN_UBI :=
+	DEVICE_DTS_CONFIG := config@oak03
 	SOC := ipq8174
-	IMAGES += factory.bin
+	IMAGES := factory.bin sysupgrade.bin
```

Comment:
You should remove `$(call Device/UbiFit)` and `KERNEL_IN_UBI :=`.
This can be simplified to `IMAGES += factory.bin`.

Reply:
I can change it this way:
```
	IMAGES := factory.bin sysupgrade.bin
	IMAGE/factory.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-ubi | linksys-image type=MX4200
	IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
```
The sysupgrade image must also be defined.